src/main/java/com/google/devtools/build/lib/runtime/BlazeRuntime.java

```diff
@@ -347,6 +343,13 @@ ProfilerStartedEvent initProfiler(
             profilePath = workspace.getOutputBase().getRelative(profileName);
             out = profilePath.getOutputStream();
           }
+        } else {
+          format =
+              options.profilePath.toString().endsWith(".gz")
+                  ? Format.JSON_TRACE_FILE_COMPRESSED_FORMAT
+                  : Format.JSON_TRACE_FILE_FORMAT;
+          profilePath = workspace.getWorkspace().getRelative(options.profilePath);
+          out = profilePath.getOutputStream(/* append= */ false, /* internal= */ true);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *out*. Execution paths that do not contain closure statements were detected.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).